src/components/ConnectToQuickbooksOnlineButton/index.tsx

```diff
@@ -1,25 +1,54 @@
+import React, {useState} from 'react';
 import Button from '@components/Button';
+import ConfirmModal from '@components/ConfirmModal';
 import useEnvironment from '@hooks/useEnvironment';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
+import {removePolicyConnection} from '@libs/actions/connections';
 import {getQuickBooksOnlineSetupLink} from '@libs/actions/connections/QuickBooksOnline';
 import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
 import type {ConnectToQuickbooksOnlineButtonProps} from './types';
 
-function ConnectToQuickbooksOnlineButton({policyID}: ConnectToQuickbooksOnlineButtonProps) {
+function ConnectToQuickbooksOnlineButton({policyID, shouldDisconnectIntegrationBeforeConnecting, integrationToDisconnect}: ConnectToQuickbooksOnlineButtonProps) {
+    const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {environmentURL} = useEnvironment();
-    const styles = useThemeStyles();
+
+    const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
+
     return (
-        <Button
-            onPress={() => Link.openLink(getQuickBooksOnlineSetupLink(policyID), environmentURL, false)}
-            text={translate('workspace.accounting.setup')}
-            style={styles.justifyContentCenter}
-        />
+        <>
+            <Button
+                onPress={() => {
+                    if (shouldDisconnectIntegrationBeforeConnecting && integrationToDisconnect) {
+                        setIsDisconnectModalOpen(true);
+                        return;
+                    }
+                    Link.openLink(getQuickBooksOnlineSetupLink(policyID), environmentURL);
+                }}
+                text={translate('workspace.accounting.setup')}
+                style={styles.justifyContentCenter}
+                small
+            />
+            {shouldDisconnectIntegrationBeforeConnecting && integrationToDisconnect && isDisconnectModalOpen && (
+                <ConfirmModal
+                    title={translate('workspace.accounting.disconnectTitle')}
+                    isVisible={isDisconnectModalOpen}
+                    onConfirm={() => {
+                        removePolicyConnection(policyID, integrationToDisconnect);
+                        Link.openLink(getQuickBooksOnlineSetupLink(policyID), environmentURL);
+                        setIsDisconnectModalOpen(false);
+                    }}
+                    onCancel={() => setIsDisconnectModalOpen(false)}
+                    prompt={translate('workspace.accounting.disconnectPrompt', CONST.POLICY.CONNECTIONS.NAME.QBO)}
+                    confirmText={translate('workspace.accounting.disconnect')}
+                    cancelText={translate('common.cancel')}
+                    danger
+                />
+            )}
+        </>
     );
 }
 
-ConnectToQuickbooksOnlineButton.displayName = 'ConnectToQuickbooksOnlineButton';
-
```

Comment:
Do we need to remove this?

Reply:
I don't see a reason to remove this, seems like a mistake. I'm going to add it back.